tests/pos/i7250.scala

```diff
@@ -0,0 +1,11 @@
+import compiletime.erasedValue
+
+sealed trait Foo
+trait A extends Foo
+trait B[H] extends Foo
+
+inline given f[T <: Foo]: T = inline erasedValue[T] match
+  case _: A => new A{}.asInstanceOf[T]
+  case _: B[a] => summon[a].asInstanceOf[T]
```

Comment:
This summon shouldn't succeed though on compile time. Starting from `println`:

```scala
// 1) initial call
f[B[Int]]

// 2) inlining `f`
inline erasedValue[B[Int]] match
  case _: A => new A{}.asInstanceOf[B[Int]]
  case _: B[a] => summon[a].asInstanceOf[B[Int]]

// 3) second case matched and inlined with a = Int
summon[Int].asInstanceOf[B[Int]]
```

`summon[Int]` should fail since there's no `Int` given in scope, shouldn't it?

```scala
scala> summon[Int]
1 |summon[Int]
  |           ^
  |no implicit argument of type Int was found for parameter x of method summon in object DottyPredef
```

Reply:
Even if it tries `Int & Foo` it still won't find anything suitable, since `f` returns `T <: Foo` and nothing about it says that it is an intersection with `Int`.

Reply:
It actually tries `f[Nothing]` and succeeds with the `A` case.

Reply:
Oh right. So the behaviour here stays the same as in the #7250 second example: if we run this test, it will fail at runtime with ClassCastException. It looks like it is the same problem as #7293, where Nothing appears in unexpected places.

Reply:
Actually I remember testing it with `inline given f[T >: Nothing <: Foo]: T`, and it still finds `f` when looking for `Nothing` (as understood from the unchanged behaviour in that case).